### ISSUE TITLE
Adding custom Calendar properties is not possible without code duplication

### ISSUE THREAD

**Issue by vazaha-nl** [markuspoerschke/iCal#383]
In my project I wanted to add an extra custom property (`X-WR-CALNAME`) to Calendars. The documentation has good info on how to do that for Events, but not for Calendars. I thought to use a similar approach but it turned out a bit less easy. To add extra custom properties I thought to extend the `getProperties` method in a subclass of src/Presentation/Factory/CalendarFactory.php. But that method is declared private so that does not work.

One could override the `createCalendar` method, which seems a bit hacky and counter intuitive, or even copy the entire class and do the modifications there. Both approaches of course do work, but are not optimal. 

Is there a reason why the `getProperties` method is declared private instead of protected?

**Comment by markuspoerschke** [comment]
The changes that you suggest makes sense in my opinion.

> Is there a reason why the getProperties method is declared private instead of protected?

The main reason is, that `private` is always the default, when I create classes. For the `EventFactory` that already changed, so yes, it makes sense to change this for the `CalendarFactory` as well.

**Comment by acirulis** [comment]
I would like to add some custom extensions from https://en.wikipedia.org/wiki/ICalendar#Calendar_extensions and found this issue. Would it be possible to make getProperties() protected, as suggested, or implement custom extension adding functionality? Thanks!

**Comment by westy92** [comment]
This is released in 2.11.0:

```php
class MyCalendarFactory extends CalendarFactory
{
  protected function getProperties(Calendar $calendar): Generator
  {
      yield from parent::getProperties($calendar);

      // Additional custom properties:
      yield new Property('X-WR-CALNAME', new TextValue('MyValue'));
  }
}
```

**Comment by vazaha-nl** [comment]
Indeed it is, thanks @markuspoerschke !
